style_guide/README.md

```diff
@@ -0,0 +1,200 @@
+    delete the feature branch.
+
+
+### Making hotfixes
+
+As shown in the diagram above, sometimes hotfixes need to be directly to the
+`main` branch. The workflow for this scenario is as follows:
+
+1. Assuming steps (1) through (3) in the previous section are already completed,
+   create a new branch named `hotfix-<description>` off of the `main` branch,
+   and commit any necessary changes following the nominal `git add`/`git commit`
+   cycle.
+2. Push the hotfix branch to your personal fork, and open two separate pull
+   requests: one that merges the hotfix branch into the
+   `IMAP-Science-Operations-Center` `imap_processing` `main` branch, and one
+   that merges the hotfix branch into the `IMAP-Science-Operations-Center`
+   `imap_processing` `dev` branch.
+3. For each of these pull requests, proceed with the nominal review & merge
+   process described in steps (9) and (10) in the previous section.
+
+
+### Keeping your fork updated
+
+You can keep your personal fork up-to-date with the
+`IMAP-Science-Operations-Cetner` `imap_processing` repository by fetching and
+pulling the `upstream` remote:
+
+    git checkout dev
+    git fetch upstream dev
+    git pull upstream/dev
+
+### Collaborating on someone else's fork
+
+To contribute to a branch on another person's personal fork, add a new `remote`
+that points to their fork, and use the nominal workflow for contributing:
+
+    git remote add <username> <remote url>
+    git fetch <username>
+    git checkout -b <branchname> <username>/<branchname>
+    # Make some changes via add/commit cycle
+    git push <username> <branchname>
+
+Python Coding
+-------------
+
+`imap_processing` code shall adhere to the [PEP8](https://www.python.org/dev/peps/pep-0008/)
+conventions save for the following exceptions:
+
+ - Lines of code need not be restricted to 79 characters.  However, it is
+encouraged to break up excessively long lines into several lines if it benefits
+the overall readability of the code
+
+ Additionally, the code shall adhere to the following special guidelines:
+
+ - Function and class definitions should be placed in alphabetical order in the
+module
+ - It is encouraged to annotate variables and functions using the
+[`typing`](https://docs.python.org/3/library/typing.html) library (see
+[PEP 483](https://www.python.org/dev/peps/pep-0483/),
+[PEP 484](https://www.python.org/dev/peps/pep-0484/), and
+[PEP 526](https://www.python.org/dev/peps/pep-0526/)).
+
+
+Documentation
+-------------
+
+`imap_processing` code shall adhere to the [PEP257](https://www.python.org/dev/peps/pep-0008/)
+and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) conventions.
+
+The following are further recommendations:
+
+- Each module should have at minimum a description, `Authors`, and `Use`
+sections.
+- Each function/method should have at minimum a description, `Parameters` (if
+necessary), and `Returns` (if necessary) sections.
```

Comment:
Are any of these verifiable through CI? You might want to go through this: https://beta.ruff.rs/docs/rules/#pydocstyle-d
and see if you want to add any of those checks to the repository.

Reply:
I opened #24 for this